agent/proxycfg/terminating_gateway.go

```diff
@@ -111,10 +112,14 @@ func (s *handlerTerminatingGateway) handleUpdate(ctx context.Context, u UpdateEv
 			svcMap[svc.Service] = struct{}{}
 
 			// Store the gateway <-> service mapping for TLS origination
-			snap.TerminatingGateway.GatewayServices[svc.Service] = *svc
+			if svc.IsEndpoint {
```

Comment:
This will change to check the `kind` field when rebased against #13217 
https://github.com/hashicorp/consul/pull/13217/files#diff-26c007c7c0d5e30fabd316255cc13ac3ffcfac5295ae8a66bcca29c197054664R610